src/vi/mimihentai/src/eu/kanade/tachiyomi/extension/vi/mimihentai/MangaDTO.kt

```diff
@@ -0,0 +1,53 @@
+package eu.kanade.tachiyomi.extension.vi.mimihentai
+
+import eu.kanade.tachiyomi.source.model.SChapter
+import keiyoushi.utils.tryParse
+import kotlinx.serialization.Serializable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+@Serializable
+class MangaDTO(
+    val data: ArrayList<Manga> = arrayListOf(),
+    val totalPage: Long,
+    val currentPage: Long,
+)
+
+@Serializable
+class Manga(
+    val id: Long,
+    val title: String,
+    val coverUrl: String,
+    val description: String,
+    val authors: ArrayList<Author> = arrayListOf(),
+    val genres: ArrayList<Genre> = arrayListOf(),
```

Comment:
Use `List<T>` instead of `ArrayList<T>`. Remove the default values since they are useless - the default `Json` instance does not fill in default values for non-nullable fields. You're supposed to have tested the extension, which implies these fields exist all the time.

Reply:
> Use `List<T>` instead of `ArrayList<T>`. Remove the default values since they are useless - the default `Json` instance does not fill in default values for non-nullable fields. You're supposed to have tested the extension, which implies these fields exist all the time.

i change to `List<T>` and toSManga() is added. So is it ready to merge?